src/move-node.js

```
@@ -20,10 +20,75 @@ const moveNode = async (toNode) => {
   return [...new Set(removedNodes)];
 };
 
+const replaceInCluster = async (nodeMap, shardMapJson) => {
+  const removedNodes = [];
+  if (!shardMapJson) {
+    throw new Error('Shard map JSON is required for multi-node migration');
+  }
+
+  const [oldNode, newNode] = Object.entries(nodeMap)[0];
+  console.log(`Migrating from ${oldNode} to ${newNode}`);
+
+  // Use the provided shard map
+  const currentDistribution = shardMapJson;
+
+  // For each shard range in the current distribution
+  console.log('Current distribution:', currentDistribution);
+  for (const [shardRange, dbNodes] of Object.entries(currentDistribution)) {
+    console.log('Shard range:', shardRange);
+    console.log('DB nodes:', dbNodes);
+    // dbNodes is an object mapping db names to node names
+    for (const [dbName, nodeName] of Object.entries(dbNodes)) {
+      console.log('DB name:', dbName);
+      console.log('Node name:', nodeName);
+      if (nodeName === oldNode) {
+        console.log(
+          `Moving shard ${shardRange} for db ${dbName} from ${oldNode} to ${newNode}`
+        );
+        const oldNodes = await moveShard.moveShard(shardRange, newNode, dbName);
+        removedNodes.push(...oldNodes);
```

So, in theory, this command should only ever move all shards from one node to another. Pushing "removedNodes" to a variable is confusing? Will there ever be other nodes? 
We should clarify this in code, by only removing the node we intend to move.
I think we should throw an error here instead, if the nodes returned by `moveShard` is not the exact node we want to remove.